pkg/networkserver/downlink.go

```diff
@@ -637,12 +638,55 @@ type downlinkPath struct {
 	*ttnpb.DownlinkPath
 }
 
-func downlinkPathsFromMetadata(mds ...*ttnpb.RxMetadata) []downlinkPath {
+func computeWantedRSSI(snr float32, channelRSSI float32) float32 {
+	wantedRSSI := channelRSSI
+	if snr <= -5.0 {
+		wantedRSSI += snr
+	} else if snr < 10.0 {
+		wantedRSSI += snr/3.0 - 10.0/3.0
+	}
+	return wantedRSSI
+}
+
+func snrMetadataComparator(mds []*ttnpb.RxMetadata) func(int, int) bool {
+	return func(i, j int) bool {
+		return mds[i].Snr >= mds[j].Snr
+	}
+}
+
+func wantedRSSIMetadataComparator(mds []*ttnpb.RxMetadata) func(int, int) bool {
+	wantedRSSI := func(k int) float32 { return computeWantedRSSI(mds[k].Snr, mds[k].ChannelRssi) }
+	return func(i, j int) bool {
+		return wantedRSSI(i) >= wantedRSSI(j)
+	}
+}
+
+var (
+	ignoreInvalidMDFeatureFlag = experimental.DefineFeature("ns.down.path.ignore_invalid_metadata", false)
+	useWantedRSSIFeatureFlag   = experimental.DefineFeature("ns.down.path.use_wanted_rssi", false)
```

Comment:
Would it be good to declare these in a `<pkg>/experiments.go` file so that we can better keep track of them? Would also be useful to add godoc or an issue reference so we know in the future what it does, where it comes from, why it is an experiment, what makes us decide to either integrate it or remove the feature as I think we don't want permanent experiments, etc.

@htdvisser did you have some thoughts on this?

Reply:
I don't think a `pkg/experiments.go` is necessary; we can already quite easily find references to `experimental.DefineFeature`.

I do agree that we should keep track of experimental features. Perhaps we can just track those in GitHub issues?

Reply:
I've opened https://github.com/TheThingsNetwork/lorawan-stack/pull/4819 regarding an issue template for experimental feature tracking.